File: src/main/java/org/activecraft/activecraftcore/exceptions/InvalidLanguageException.java
package org.activecraft.activecraftcore.exceptions;

import org.activecraft.activecraftcore.ActiveCraftPlugin;
import org.activecraft.activecraftcore.messages.Language;
import lombok.Getter;
import org.activecraft.activecraftcore.ActiveCraftPlugin;

@Getter
public class InvalidLanguageException extends ActiveCraftException{

    private Language language;
    private ActiveCraftPlugin plugin;

    public InvalidLanguageException(String message, Language language, ActiveCraftPlugin plugin) {
        super(message);
        this.language = language;
        this.plugin = plugin;
    }

    public InvalidLanguageException(Language language, ActiveCraftPlugin plugin) {
        this("Invalid language.", language, plugin);
    }
}

File: src/main/java/org/activecraft/activecraftcore/exceptions/InvalidLanguageException.kt
package org.activecraft.activecraftcore.exceptions

import org.activecraft.activecraftcore.ActiveCraftPlugin
import org.activecraft.activecraftcore.messages.Language

class InvalidLanguageException @JvmOverloads constructor(
    val language: Language?,
    val plugin: ActiveCraftPlugin,
    message: String? = "Invalid language."
) : ActiveCraftException(message)